server/sonar-server/src/main/java/org/sonar/server/measure/ws/SearchAction.java

```diff
@@ -142,18 +145,26 @@ private SearchRequest setRequest() {
 
     private List<MetricDto> searchMetrics() {
       requireNonNull(request);
-      return dbClient.metricDao().selectByKeys(dbSession, request.getMetricKeys());
+      List<MetricDto> dbMetrics = dbClient.metricDao().selectByKeys(dbSession, request.getMetricKeys());
+      List<String> metricKeys = dbMetrics.stream().map(MetricDto::getKey).collect(Collectors.toList());
+      checkRequest(request.getMetricKeys().size() == dbMetrics.size(), "The following metrics are not found: %s",
+        String.join(", ", difference(request.getMetricKeys(), metricKeys)));
+      return dbMetrics;
```

Comment:
- the user has provided the information
- the probability that the request is incorrect is very high

-> we should fail with a proper message